core/templates/domain/topic/topic-rights-backend-api.service.ts

```diff
@@ -16,39 +16,58 @@
  * @fileoverview Service to change the rights of topic in the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import cloneDeep from 'lodash/cloneDeep';
+
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service.ts';
+import { TopicDomainConstants } from
+  'domain/topic/topic-domain.constants';
+import { TopicRightsBackendDict, TopicRights }
+  from 'domain/topic/topic-rights.model.ts';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TopicRightsBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolation: UrlInterpolationService) {}
 
-require('domain/topic/topic-domain.constants.ajs.ts');
-
-angular.module('oppia').factory('TopicRightsBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService',
-  'TOPIC_RIGHTS_URL_TEMPLATE',
-  function(
-      $http, $q, UrlInterpolationService, TOPIC_RIGHTS_URL_TEMPLATE) {
     // Maps previously loaded topic rights to their IDs.
-    var topicRightsCache = {};
-
-    var _fetchTopicRights = function(
-        topicId, successCallback, errorCallback) {
-      var topicRightsUrl = UrlInterpolationService.interpolateUrl(
-        TOPIC_RIGHTS_URL_TEMPLATE, {
+    private _topicRightsCache = {};
+
+    private _fetchTopicRights(
+        topicId: string,
+        successCallback: (value?: Object | PromiseLike<Object>) => void,
```

Comment:
Can you change this from PromiseLike to Promise and change the type from Object to the actual type we are using (I think the type is TopicRight, check once). Can you also do that in `_setTopicStatus` function?
```suggestion
        successCallback: (value?: Object | Promise<TopicRights>) => void,
```

Reply:
Done!